sdcflows/interfaces/bspline.py

```diff
@@ -124,27 +131,50 @@ class BSplineApprox(SimpleInterface):
 
     def _run_interface(self, runtime):
         from sklearn import linear_model as lm
-        from scipy.sparse import vstack as sparse_vstack
+
+        # Output name baseline
+        out_name = fname_presuffix(
+            self.inputs.in_data, suffix="_field", newpath=runtime.cwd
+        )
 
         # Load in the fieldmap
         fmapnii = nb.load(self.inputs.in_data)
-        data = fmapnii.get_fdata(dtype="float32")
+        zooms = fmapnii.header.get_zooms()
 
-        # Generate the output naming base
-        out_name = fname_presuffix(
-            self.inputs.in_data, suffix="_field", newpath=runtime.cwd
+        # Get a mask (or define on the spot to cover the full extent)
+        masknii = (
+            nb.load(self.inputs.in_mask)
+            if isdefined(self.inputs.in_mask)
+            else None
         )
-        # Create a copy of the header for use below
-        hdr = fmapnii.header.copy()
-        hdr.set_data_dtype("float32")
 
+        need_resize = np.any(np.array(zooms) < self.inputs.zooms_min)
+        if need_resize:
+            from sdcflows.utils.tools import resample_to_zooms
+
+            zooms_min = self.inputs.zooms_min
+
+            with suppress(TypeError):
+                zooms_min = [float(zooms_min)] * 3
+
+            LOGGER.info(
+                "Resampling image with resolution exceeding 'zooms_min' "
+                f"({'x'.join(str(s) for s in zooms)})."
+            )
+            fmapnii = resample_to_zooms(fmapnii, zooms_min)
+
+            if masknii is not None:
+                masknii = resample_to_zooms(masknii, zooms_min)
```

Comment:
I still don't think we want to upsample in cases of anisotropic voxels. The goal is to downsample when convenient.

```suggestion
            target_zooms = np.maximum(zooms, self.inputs.zooms_min)

            LOGGER.info(
                f"Resampling image with resolution exceeding {zooms_min} "
                f"({'x'.join(str(s) for s in zooms)})."
            )
            fmapnii = resample_to_zooms(fmapnii, target_zooms)

            if masknii is not None:
                masknii = resample_to_zooms(masknii, target_zooms)
```

If we want to force isotropy, that might be a special case that deserves its own parameter.

Reply:
Not sure what you are pointing out as a problem. The PR just adds a way of imposing an upper bound to the resolution. The upper bound can be iso- or anisotropic, depending on what you pass into `zooms_min`.

What I am arguing is that, if you want specific zooms (e.g., following a 4:1 ratio), then you set up an interface to calculate them, and you pass them in. And if you want something even more specific (e.g., smoothing) then, you do the resampling outsize and set ``zooms_min = 0.0`` to ensure there's no further resampling.

Reply:
Right now, if I set the value to 1.95 and I have voxels that are 1.94x1.94x3, I will get out 1.95x1.95x1.95, which down-samples in two dimensions and up-samples in the other. If I have voxels that are 1.98x1.98x3, I get 1.98x1.98x3.

The proposal above would change it to only down-sample, so 1.94x1.94x3 would get changed to 1.95x1.95x3, which seems like less of a qualitatively different behavior when the threshold is crossed.

Reply:
Okay, I think I understand your point. Basically something like:

```Python
final_zooms = np.max(zip(old_zooms, new_zooms))
```

Reply:
Yes. I had `target_zooms = np.maximum(zooms, self.inputs.zooms_min)`